build.gradle

```diff
@@ -75,6 +75,10 @@ dependencies {
 }
 
 jacocoTestReport {
+    doFirst {
+        classDirectories = classDirectories.collect { fileTree(it) { exclude '**/*$*_closure*' } }.sum()
+    }
+
```

Comment:
Do **not** do this.
Or at least tell me how this brings "better coverage metrics."
It does not at all bring better metrics.
The only thing you achieve with this is the same you achieved the first time you added it before I first contributed and removed it, it is self-deception, nothing else.

You might *think* at first sight that the metrics report looks cleaner if the closures are not listed as own classes, especially as they are also included in the full classes. But this is non-sense and the same as with anonymous classes. If you cut them out, from recording and reporting they are simply not considered at all and thus you raise the coverage percentages. But for real values you have to leave them in where they belong.

I've also prepared some graphics that show you some of the differences, upper half is this branch as you committed it, lower half the same with only the excludes removed.

## Overview:
Watch the numbers here. 300 vs. 432 branches, 302 vs. 361 Lines, 12 vs. 48 Classes, ...
![overview](https://cloud.githubusercontent.com/assets/325196/22575927/060b8f1a-e9b9-11e6-8c5e-20d404c62b59.png)

## Package:
Of course the closures look a bit strange here and you might not be able to easily identify them in the source, but I found no way to leave them out in only this particular view. But it surely could be done with a custom CSS or something like that if that is what bugs you most, but even then it would not really be correct, as their values are *not* included in the enclosing classes values, they are completely separate, besides that they share the same source file.
![package](https://cloud.githubusercontent.com/assets/325196/22575984/49b1c45a-e9b9-11e6-9cea-19bf46905727.png)

## File:
Here you see the difference best. As you can see in the upper version almost nothing is even considered (white), in the lower version all code lines are covered, as the closure which makes most of the file here is included in the metrics and report.
![file](https://cloud.githubusercontent.com/assets/325196/22576062/c6baf502-e9b9-11e6-9d2b-1cd1fc95fd42.png)

If your problem was that you didn't get the 0 missed branches, lines and classes, don't forget to set the project property `disableGroovyOptimizations` when running tests for getting proper covering metrics, otherwise the Groovy magic does create too much stuff that is non-obvious and not really testable. That's also why it is enabled in the circleci test task, but not in the deployment task. This means the sources will need to be recompiled though they were compiled already for the tests, but you get proper coverage metrics out.